### PR TITLE
chore: caching requests to optmimize load times and avoid rate limits

### PR DESCRIPTION
**Pull request by ximxim**
## Description

- Remove band aid solutions for both tweets and arweave fetches
- Added caching to Arweave, Twitter and Mint.Fun calls

## Ticket(s)

- [69](https://github.com/LazerTechnologies/onchain-summer/issues/69)

## Deploy Notes

Notes regarding deployment of the contained body of work. These should note any
db migrations, nginx routes, infrastructure changes, and anything that must be done before deployment.

- [ ] Need to add environment variables
  - `ENV_VAR=`

### CODE REVIEW

**Review by lukasrosario — APPROVED** [review]

File: src/utils/getURL.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export const getURL = () => {
```
Comment (mdehoog) [inline src/utils/getURL.ts:1]: can we use this for the metadata fields too? e.g. anywhere that `https://onchainsummer.xyz` is hard coded

**Review by wilsoncusack — APPROVED** [review]

**Review by wilsoncusack — CHANGES_REQUESTED** [review]
waiting for password

File: src/utils/getURL.ts (hunk 1 above)
Comment (ximxim) [reply src/utils/getURL.ts:1]: Good point, resolved.

**Comment by cb-heimdall** [comment]
Review Error for wilsoncusack @ 2023-08-10 17:44:04 UTC
User failed mfa authentication, see [go/mfa-help](http://go/mfa-help)

**Comment by cb-heimdall** [comment]
Review Error for wilsoncusack @ 2023-08-10 17:45:12 UTC
User failed mfa authentication, see [go/mfa-help](http://go/mfa-help)

**Review by wilsoncusack — APPROVED** [review]

**Review by wilsoncusack — APPROVED** [review]

**Review by wilsoncusack — APPROVED** [review]

**Comment by cb-heimdall** [comment]
Review Error for wilsoncusack @ 2023-08-10 19:34:29 UTC
User failed mfa authentication, see [go/mfa-help](http://go/mfa-help)

**Pull request merged**